assets/js/googlesitekit/widgets/util/get-widget-layout.js

```diff
@@ -19,49 +19,29 @@
 /**
  * Internal dependencies
  */
-import { WIDGET_WIDTHS } from '../datastore/constants';
-
-const WIDTH_GRID_COUNTER_MAP = {
-	[ WIDGET_WIDTHS.QUARTER ]: 3,
-	[ WIDGET_WIDTHS.HALF ]: 6,
-	[ WIDGET_WIDTHS.FULL ]: 12,
-};
-
-const WIDTH_GRID_CLASS_MAP = {
-	[ WIDGET_WIDTHS.QUARTER ]: [
-		'mdc-layout-grid__cell',
-		'mdc-layout-grid__cell--span-2-phone',
-		'mdc-layout-grid__cell--span-3-desktop',
-		'mdc-layout-grid__cell--span-4-tablet',
-	],
-	[ WIDGET_WIDTHS.HALF ]: [
-		'mdc-layout-grid__cell',
-		'mdc-layout-grid__cell--span-6-desktop',
-		'mdc-layout-grid__cell--span-8-tablet',
-	],
-	[ WIDGET_WIDTHS.FULL ]: [
-		'mdc-layout-grid__cell',
-		'mdc-layout-grid__cell--span-12',
-	],
-};
+import { WIDTH_GRID_COUNTER_MAP, WIDTH_GRID_CLASS_MAP } from './constants';
 
 /**
  * Adjusts class names to better fit into the current row knowing that the default sizes don't fill the row completely.
  *
  * @since 1.21.0
  *
- * @param {Array.<string>} classNames Current class names.
- * @param {number}         counter    Current counter.
- * @return {Array} Array where the first element is the updated list of class names
- *                 and the second element is the resulting counter after the update.
+ * @param {Array.<string>} classNames   Current class names.
+ * @param {Array.<number>} columnWidths Current column widths.
+ * @param {number}         counter      Current counter.
+ * @return {Array} Array where the first element is the updated list of class
+ *                 names, the second element is the updated list of column
+ *                 widths, and the third element is the resulting counter
+ *                 after the update.
  */
-function resizeClasses( classNames, counter ) {
+function resizeClasses( classNames, columnWidths, counter ) {
 	// Safeguard: counter must always be 9 for this to work.
 	if ( counter !== 9 ) {
-		return [ classNames, counter ];
+		return [ classNames, columnWidths, counter ];
```

Comment:
Maybe we should change this to return an object now that we're adding another value to the return?

Reply:
Normally I would agree, but I think here it makes sense to keep it as array:
* The return value is basically a "filtered" version of each function parameter, so having them in an array is intuitive from that perspective.
* Destructuring the values into existing variables like we currently do from an array when calling `resizeClasses` work, but with an object it wouldn't be valid (I wasn't aware of this JS detail before). We would need to first assign it to an object and then write each property's value into the respective variable.

Reply:
I think you mean this:

![image](https://user-images.githubusercontent.com/1621608/105489162-7fefe500-5cbb-11eb-9f48-8cba24e0f436.png)

If so, it is possible (and we do it in a few places in the codebase), it just requires the added parentheses 👍